app/src/main/java/com/example/bigowlapp/service/LocationTrackingExpiredAlarmReceiver.java

```diff
@@ -1,19 +1,67 @@
 package com.example.bigowlapp.service;
 
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
 import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
+import android.os.Build;
+import android.os.PowerManager;
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
 
+import com.example.bigowlapp.R;
 import com.example.bigowlapp.utils.PeriodicLocationCheckAlarmManager;
+import com.example.bigowlapp.utils.ScheduledLocationTrackingManager;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 
 /**
  * This will run when location tracking has expired for a user as a result of not getting
  * to a location at the expected time.
  */
 public class LocationTrackingExpiredAlarmReceiver extends BroadcastReceiver {
+    private static final String CHANNEL_ID = "Missed Schedule";
+
     @Override
     public void onReceive(Context context, Intent intent) {
         PeriodicLocationCheckAlarmManager locationCheckAlarmManager = new PeriodicLocationCheckAlarmManager(context);
         locationCheckAlarmManager.cancelPeriodicLocationCheck();
+
+        sendMissedScheduleNotification(context);
+        PowerManager pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
+        if (!pm.isInteractive()) {
+            PowerManager.WakeLock wl = pm.newWakeLock(PowerManager.SCREEN_DIM_WAKE_LOCK | PowerManager.ACQUIRE_CAUSES_WAKEUP, "myApp:notificationLock");
+            wl.acquire(3000);
+        }
+
+
+        Date date = new Date(System.currentTimeMillis()-ScheduledLocationTrackingManager.DEFAULT_TRACKING_EXPIRE_TIME_MILLIS);
+        DateFormat df = new SimpleDateFormat("MM/dd/yyyy HH:mm");
```

Comment:
We already have a date  and time formatter in DateTimeFormatter.java. You should use it instead to remove duplicated implementation from here.

The formatter in that version doesn't combine both date and time in one call, so you can also create a method that does that in DateTimeFormatter.java, rather than doing a call to both `dateFormatter()` and `timeFormatter()`